### PR TITLE
Adds a chameleon EVA suit to the uplink

### PR DESCRIPTION
**Pull request by Arendian**
## About the PR

This PR adds a chameleon EVA suit and helmet to the uplink.
The suit has the exact same stats as the normal syndicate EVA(no armor, 20% slowdown) the helmet also has no armor value.
The bundle contains the following:
Chameleon duffel bag
Chameleon EVA helmet
Chameleon EVA suit
Chameleon gas mask
Double emergency oxygen tank

**Media**

image
image
image

- [X] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

**Changelog**

:cl: Dygon
- add: Syndicate researchers have finally found a way to fit their chameleon technology into an EVA suit. The chameleon EVA bundle is now available in the uplink.

### CODE REVIEW

**Comment by 0x6273** [comment]
Seems kinda overpowered for just 6 TC. Also, the code has formatting issues in some places.

**Comment by Arendian** [comment]
> Seems kinda overpowered for just 6 TC. Also, the code has formatting issues in some places.

The normal suit is 2 tc so I thought 4 tc for a the chameleon ability sounded acceptable, the normal chameleon clothing is only 6 tc too and contains way more stuff. What price do you think is more representable for it's function? 8tc?(same price as bloodred)

**Comment by deltanedas** [comment]
8 tc can get you an emag, and thus a real hardsuit
why pretend to be cap in space with chameleon eva when you can just get the real thing, it will probably be left in the storage anyway

**Comment by Arendian** [comment]
> 8 tc can get you an emag, and thus a real hardsuit
> why pretend to be cap in space with chameleon eva when you can just get the real thing, it will probably be left in the storage anyway

This suit would make nuclear operatives infiltrating a lot more viable since you have no timeframe where you are in your bloodred when entering the station. Also the current chameleon does not have the option to imitate any hardsuit at all, so this one would make it possible to infiltrate engineering and blowing the ame for example or turning off grav without looking too suspicious. Or switch yourself to a security hardsuit and walk around undercover during an assault without worrying about the spacing of your nuclear operative colleagues.

**Comment by KaiShibaa** [comment]
I feel like chameleon should just be tied to suits or regular clothing and not actual EVA protection suit (or adding chameleon to everything). If the point of it is to be stealthy you can kinda already do that with the normal chameleon gear wearing Captain's hardsuit for instance and faking that you're wearing it (by not going into space) 

If you space an area as a nukie, I think you've already lost the element of stealth seeing that people already know you're onboard.

File: Content.Shared/Clothing/EntitySystems/SharedChameleonClothingSystem.cs [hunk 1]
```diff
@@ -87,4 +87,22 @@ public bool IsValidTarget(EntityPrototype proto, SlotFlags chameleonSlot = SlotF
 
         return true;
     }
+    public bool IsValidEVATarget(EntityPrototype proto, SlotFlags chameleonSlot = SlotFlags.NONE)
```
Comment (0x6273) [inline Content.Shared/Clothing/EntitySystems/SharedChameleonClothingSystem.cs:90]:
```suggestion
    }
    
    public bool IsValidEVATarget(EntityPrototype proto, SlotFlags chameleonSlot = SlotFlags.NONE)
```

**Review by 0x6273 — COMMENTED** [review]
> The normal suit is 2 tc so I thought 4 tc for a the chameleon ability sounded acceptable, the normal chameleon clothing is only 6 tc too and contains way more stuff. What price do you think is more representable for it's function? 8tc? (same price as bloodred) It would make this the suit for a stealth mission while a bloodred would be for a loud one.

Yeah 8 is probably fine. It can always be tuned later.

**Comment by Arendian** [comment]
> I feel like chameleon should just be tied to suits or regular clothing and not actual EVA protection suit (or adding chameleon to everything). If the point of it is to be stealthy you can kinda already do that with the normal chameleon gear wearing Captain's hardsuit for instance and faking that you're wearing it (by not going into space). You can already do the things you've used as an example with the normal chameleon suit.
> 
> If you space an area as a nukie, I think you've already lost the element of stealth seeing that people already know you're onboard.

The current chameleon gear is unable to disguise as hardsuits(except rd one), only softsuits.
image


And if you space the area in disguise people would just shout "an engineer just exploded x area" instead of "nukies just exploded x area"

**Comment by Myakot** [comment]
8 tc is way too expensive. Don't forget this suit's parent is default syndie eva. This suit **doesn't** protect you in open space and the cold temperatures will get to you.
also rip slimes with oxy tanks in bundles, as always.

**Comment by KaiShibaa** [comment]
> 8 tc is way too expensive. Don't forget this suit's parent is default syndie eva. This suit **doesn't** protect you in open space and the cold temperatures will get to you. also rip slimes with oxy tanks in bundles, as always.

You're probably thinking of the emergency suits. Syndie EVA does protect you from space.

Price is higher because it literally allows you to change the look of the suit on a whim. If it were the same price as the regular Syndie EVA bundle, this would just be better to grab than that.

**Review by AJCM-git — CHANGES_REQUESTED** [review]
This unnecessarily repeats a lot of code, you need to add a new EntityWhitelist datafield to the chamaleon component and tweak the methods to support it

**Comment by Emisse** [comment]
personally not a fan of space proof chameleon suit

**Review by crazybrain23 — CHANGES_REQUESTED** [review]
~~Ironically, you've missed the EVA suit (it's in the `softsuits.yml`)~~
I might be wrong here, I have no idea how you did it without tags?

**Comment by AJCM-git** [comment]
re-open if you come back to this

**Pull request closed without merging**